compiler/rustc_errors/src/diagnostic_builder.rs

```diff
@@ -184,16 +184,17 @@ impl<'a> DiagnosticBuilder<'a> {
         self.cancel();
     }
 
-    /// Adds a span/label to be included in the resulting snippet.
+    /// Appends a labeled span to the diagnostic.
     ///
-    /// This is pushed onto the [`MultiSpan`] that was created when the diagnostic
-    /// was first built. That means it will be shown together with the original
-    /// span/label, *not* a span added by one of the `span_{note,warn,help,suggestions}` methods.
+    /// Labels are used to convey additional context for the diagnostic's primary span. They will
+    /// be shown together with the original diagnostic's span, *not* with spans added by
+    /// `span_note`, `span_help`, etc. If the primary span is not displayable, labels will not be
+    /// displayed either.
```

Comment:
When is a span "not displayable"? Would be good to have an example.

Reply:
When they are `DUMMY_SP` or the source code for them isn't found (external crates or std on certain deployments where remap path isn't enabled).